Add Auth Generate JWT example (#2380)
ably/docs: examples/auth-generate-jwt/page.md

# Request an Ably JWT to authenticate clients

> Use an Ably JWT to securely authenticate your users.

Reviewer: ```suggestion
Use a JWT (JSON Web Token) to securely authenticate your users.
```

> Ably JWT enable you to grant users access to the Ably platform. They are used to issue time-limited credentials that securely authorize users, and manage which resources each user has access to.

Reviewer: ```suggestion
Ably [JWTs](https://jwt.io/) enable you to grant users access to the Ably platform. They are used to issue time-limited credentials that securely authorize users, and manage which resources each user has access to. You can use a JWT-compliant library to create an Ably JWT without needing to request a token from Ably.
```

Authentication is implemented using [Ably Pub/Sub](https://ably.com/docs/auth). The Pub/Sub SDK provides the authentication mechanism that is utilized by all Ably products. It provides a set of flexible APIs capable of building any realtime application and is powered by Ably's reliable and scalable platform.

## Resources

// React brief (Only visible if viewing the React example)

This example uses a frontend and a backend client. The frontend is a React application and the backend is an Express Node.js application. Both of these applications use Ably Pub/Sub.

The server application generates an Ably JWT using the Ably API key with the key name part of the API key as `kid` in the headers. The API secret part is used to encrypt the JWT. The Ably JWT that is returned is then used by the frontend client as authentication with Ably.

> Find out more about [authentication](https://ably.com/docs/auth/token?lang=javascript#standard).

Reviewer: This needs to be updated to link to JWT in both places.
